File: Atividade_3/ContaBancaria.go
package main

import "fmt"

type ContaBancaria struct {
    Titular string
    saldo   float64 
}

func (c *ContaBancaria) Depositar(valor float64) {
    if valor > 0 {
        c.saldo += valor
        fmt.Printf("Depósito de R$ %.2f realizado com sucesso!\n", valor)
    } else {
        fmt.Println("Valor inválido para depósito.")
    }
}

func (c *ContaBancaria) Sacar(valor float64) {
    if valor > 0 && valor <= c.saldo {
        c.saldo -= valor
        fmt.Printf("Saque de R$ %.2f realizado com sucesso!\n", valor)
    } else {
        fmt.Println("Saldo insuficiente ou valor inválido.")
    }
}

func (c *ContaBancaria) ExibirSaldo() {
    fmt.Printf("Saldo atual: R$ %.2f\n", c.saldo)
}

func main() {
    conta := ContaBancaria{Titular: "João", saldo: 1000}
    conta.Depositar(500)
    conta.Sacar(300)
    conta.ExibirSaldo()
}

File: Atividade_3/ContaBancaria.py
class ContaBancaria:
    def __init__(self, titular, saldo=0):
        self.titular = titular
        self.__saldo = saldo 

    def depositar(self, valor):
        if valor > 0:
            self.__saldo += valor
            print(f"Depósito de R$ {valor} realizado com sucesso!")
        else:
            print("Valor inválido para depósito.")

    def sacar(self, valor):
        if 0 < valor <= self.__saldo:
            self.__saldo -= valor
            print(f"Saque de R$ {valor} realizado com sucesso!")
        else:
            print("Saldo insuficiente ou valor inválido.")

    def exibir_saldo(self):
        print(f"Saldo atual: R$ {self.__saldo}")

conta = ContaBancaria("João")
conta.depositar(1000)
conta.sacar(500)
conta.exibir_saldo()
